cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/web/groups/GroupsActionFactory.java

```diff
@@ -109,12 +109,16 @@ public AbstractGroupAction createAction(HttpServletRequest request)
                 {
                     action = new GetGroupNamesAction();
                 }
-                else if (segments.length == 1)
+                else if ((segments.length == 1) && (segments[0].equals("list")))
+                {
+                    action = new GetGroupsListAction();
+                }
+                else if ((segments.length == 1) && (!segments[0].equals("list")))
                 {
                     String groupName = segments[0];
-                    action = new GetGroupAction(groupName);
+                    action = new GetGroupAction(groupName);                
                 }
-
+                        
                 if (action != null)
```

Comment:
I think it would be better if the GetGroupListAction was invoked if the path was /groups (ie segments.length==0) and there was a parameter "detail" with the value of "max".  This more closely reflects the IVOA approach to REST calls.